webui/react/src/pages/Authentication.tsx

```diff
@@ -0,0 +1,156 @@
+import { Button, Form, Input } from 'antd';
+import { message } from 'antd';
+import axios from 'axios';
+import queryString from 'query-string';
+import React, { useEffect, useState } from 'react';
+import { useHistory, useLocation } from 'react-router';
+import { Redirect } from 'react-router-dom';
+
+import Icon from 'components/Icon';
+import Logo, { LogoTypes } from 'components/Logo';
+import Spinner from 'components/Spinner';
+import Auth, { updateAuth } from 'contexts/Auth';
+import handleError, { ErrorType } from 'ErrorHandler';
+import { crossoverRoute, isCrossoverRoute } from 'routes';
+import { isLoginFailure, login, logout } from 'services/api';
+import { Credentials } from 'types';
+
+import css from './Authentication.module.scss';
+
+const DEFAULT_REDIRECT = '/det/dashboard';
+
+type WithSearch<T> = T & {location: {search: string}};
+interface Queries {
+  redirect?: string;
+}
+
+interface FromValues {
+  password?: string;
+  username?: string;
+}
+
+const Authentication: React.FC<WithSearch<{}>> = (props: WithSearch<{}>) => {
+  const history = useHistory();
+  const location = useLocation();
+  const auth = Auth.useStateContext();
+  const setAuth = Auth.useActionContext();
+  const [ isLoading, setIsLoading ] = useState(true);
+  const [ badCredentials, setBadCredentials ] = useState(false);
+  const [ canSubmit, setCanSubmit ] = useState(false);
+
+  const queries: Queries = queryString.parse(props.location.search);
```

Comment:
shouldn't queryString just parse `location.search` from the `useLocation` hook already defined above? I don't think `WithSearch` is needed.

Reply:
you're right! I think I had added those in separate steps.